Support range select for amending commit attributes

AzraelSec opened the pull request:
- **PR Description**

This PR makes it possible for users to select a range of commits from the commits view to amend their attributes (set/reset author, add co-author), the same way it's already possible to do for a single commit.

It closes #3273. 

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go generate ./...`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [ ] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [ ] Docs have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

<!--
Be sure to name your PR with an imperative e.g. 'Add worktrees view'
see https://github.com/jesseduffield/lazygit/releases/tag/v0.40.0 for examples
-->

stefanhaller commented on pkg/gui/controllers/local_commits_controller.go:
```diff
@@ -695,60 +695,68 @@ func (self *LocalCommitsController) amendTo(commit *models.Commit) error {
 
-func (self *LocalCommitsController) amendAttribute(commit *models.Commit) error {
+func (self *LocalCommitsController) canAmend(_ *models.Commit) *types.DisabledReason {
+	if !self.isHeadCommit(self.context().GetSelectedLineIdx()) && self.isRebasing() {
+		return &types.DisabledReason{Text: self.c.Tr.AlreadyRebasing}
+	}
+
+	return nil
```
This could be shortened to
```go
	idx := self.context().GetSelectedLineIdx()
	return self.canAmendRange(self.c.Model().Commits, idx, idx)
```

stefanhaller commented on pkg/gui/controllers/local_commits_controller.go:
```diff
@@ -1188,7 +1196,11 @@ func (self *LocalCommitsController) markAsBaseCommit(commit *models.Commit) erro
+}
+
+func (self *LocalCommitsController) isSelectedHeadCommit() bool {
 	return models.IsHeadCommit(self.c.Model().Commits, self.context().GetSelectedLineIdx())
```
Minor nitpick: I'd change this to
```go
	return self.isHeadCommit(self.context().GetSelectedLineIdx())
```

stefanhaller reviewed (commented):
Sorry for only reviewing this now.

This looks great, very well done. I have two small suggestions below, nothing major.

One minor concern is that the commit is pretty large, this makes reviewing a bit arduous. I don't have really good ideas how to split it up though. One thing that could be extracted into a separate commit is renaming isHeadCommit to isSelectedHeadCommit (in preparation for adding isHeadCommit that takes an index), but that doesn't help very much; so nothing to act on, really.

AzraelSec commented:
Thanks for the feedback, @stefanhaller, and sorry for having raised such a huge PR. I was honestly unsure of how to split it in a good way. I addressed your comments, really nice catch! The original plan was to do what you suggested from the beginning, but I must have missed it on the way.

stefanhaller commented:
Awesome, thanks. Great work!
